monai/transforms/utility/array.py

```diff
@@ -316,6 +323,36 @@ def __call__(self, img) -> torch.Tensor:
         return torch.as_tensor(img)
 
 
+class EnsureType(Transform):
+    """
+    Ensure the input data to be a PyTorch Tensor or numpy array, support: `numpy array`, `PyTorch Tensor`,
+    `float`, `int`, `bool`. If passing a dictionary, list or tuple, recursively check every item and ensure
+    it to be expected data type.
+
+    Args:
+        dtype: target data type to convert, should be "tensor" or "numpy".
+
+    """
+
+    def __init__(self, dtype: str = "tensor") -> None:
+        dtype = dtype.lower()
+        if dtype not in ("tensor", "numpy"):
+            raise ValueError("`dtype` must be 'tensor' or 'numpy'.")
+
+        self.dtype = dtype
+
+    def __call__(self, data):
+        """
+        Args:
+            data: input data can be PyTorch Tensor, numpy array, list, dictionary, int, float, bool, str, etc.
+                will ensure Tensor, Numpy array, float, int, bool as Tensors or numpy arrays, strings and
+                objects keep the original. for dictionay, list or tuple, ensure every item as expected type
+                if applicable.
+
+        """
+        return convert_to_tensor(data) if self.dtype == "tensor" else convert_to_numpy(data)
```

Comment:
should probably throw something if self.dtype is unknown. Also should probably be `type`, not `dtype` as the latter is normally e.g., `np.float32`.

Reply:
Hi @rijobro , I think `type` is python internal operator we can't use it as an arg, right?

Thanks.

Reply:
Ok, how about `data_type`?

Reply:
@wyli @ericspod , what do you guys think?
Thanks.

Reply:
sure, data_type looks good... or simply `t`: `EnsureType(t='tensor')`

Reply:
Thanks for your discussion.
`t` seems too simple and a little bit confusing, changed to `data_type` in the latest commit.

Thanks.